mat/solve.go

```diff
@@ -104,24 +104,37 @@ func (m *Dense) Solve(a, b Matrix) error {
 }
 
 // SolveVec finds a minimum-norm solution to a system of linear equations defined
-// by the matrix a and the right-hand side vector b. If A is singular or
+// by the matrix a and the right-hand side column vector b. If A is singular or
 // near-singular, a Condition error is returned. Please see the documentation for
 // Dense.Solve for more information.
-func (v *VecDense) SolveVec(a Matrix, b *VecDense) error {
-	if v != b {
-		v.checkOverlap(b.mat)
+func (v *VecDense) SolveVec(a Matrix, b Vector) error {
+	if _, bc := b.Dims(); bc != 1 {
```

Comment:
The other methods check the dims of `b` against the matrix. Should it be done here too?

Reply:
This method calls `(*Dense).Solve` and the dims are checked there. The `bc` check is special for this to ensure that the user has passed in a column vector.